packages/react-components/react-dialog/src/components/DialogSurface/useDialogSurface.ts

```diff
@@ -16,39 +27,81 @@ import { isEscapeKeyDismiss } from '../../utils';
+  const open = useDialogContext_unstable(ctx => ctx.open);
+  const requestOpenChange = useDialogContext_unstable(ctx => ctx.requestOpenChange);
   const dialogTitleID = useDialogContext_unstable(ctx => ctx.dialogTitleID);
   const dialogBodyID = useDialogContext_unstable(ctx => ctx.dialogBodyID);
-  const requestOpenChange = useDialogContext_unstable(ctx => ctx.requestOpenChange);
 
-  const { modalAttributes } = useModalAttributes({ trapFocus: modalType !== 'non-modal' });
+  const handledBackdropClick = useEventCallback((event: React.MouseEvent<HTMLDivElement>) => {
+    if (isSlot(props.backdrop)) {
+      props.backdrop.onClick?.(event);
+    }
+    if (modalType === 'modal' && !event.isDefaultPrevented()) {
+      requestOpenChange({
+        event,
+        open: false,
+        type: 'backdropClick',
+      });
+    }
+  });
 
-  const handleRootKeyDown = useEventCallback((event: React.KeyboardEvent<HTMLDivElement>) => {
+  const handleKeyDown = useEventCallback((event: React.KeyboardEvent<DialogSurfaceElementIntersection>) => {
     props.onKeyDown?.(event);
-    if (isEscapeKeyDismiss(event, modalType)) {
-      requestOpenChange({ event, open: false, type: 'escapeKeyDown' });
+
+    if (
+      event.key === Escape &&
+      // `non-modal` should always have Escape key handling
+      // `modal` should only be handled in the case of non native dialog
+      (modalType === 'non-modal' || modalType === 'modal') &&
+      !event.isDefaultPrevented()
+    ) {
+      requestOpenChange({
+        event,
+        open: false,
+        type: 'escapeKeyDown',
+      });
       event.preventDefault();
     }
   });
 
   return {
     components: {
+      backdrop: 'div',
       root: 'div',
     },
-    root: getNativeElementProps(as, {
-      ref: useMergedRefs(ref, contentRef),
-      'aria-modal': modalType !== 'non-modal',
-      'aria-describedby': dialogBodyID,
-      'aria-labelledby': props['aria-label'] ? undefined : dialogTitleID,
-      role: modalType === 'alert' ? 'alertdialog' : 'dialog',
-      ...props,
-      ...modalAttributes,
-      onKeyDown: handleRootKeyDown,
+    backdrop: resolveShorthand(backdrop, {
+      required: open && modalType !== 'non-modal',
+      defaultProps: {
+        'aria-hidden': 'true',
+        onClick: handledBackdropClick,
+      },
     }),
+    root: getNativeElementProps(
+      as ?? 'div',
+      resolveShorthand(props as DialogSurfaceRootSlotProps, {
+        required: true,
+        defaultProps: {
+          onKeyDown: handleKeyDown,
+          'aria-describedby': dialogBodyID,
+          'aria-labelledby': props['aria-label'] ? undefined : dialogTitleID,
+          'aria-modal': modalType !== 'non-modal',
+          role: props.role ?? (modalType === 'alert' ? 'alertdialog' : 'dialog'),
+          ref: useMergedRefs(ref, dialogRef) as React.Ref<DialogSurfaceElementIntersection>,
+        },
+      }),
+    ),
   };
 };
+
+// TODO: move this to @fluentui/react-utilities
```

Comment:
you could just do this, a teams-prg review covers `react-utilities`

Reply:
This has been done already actually... the method is called `isResolvedShorthand`. I guess this is just old code I've copied from previous PR